src/NeoSharp.Core/Messaging/Handlers/BlockMessageHandler.cs

```diff
@@ -50,10 +53,15 @@ public override bool CanHandle(Message message)
         public override async Task Handle(BlockMessage message, IPeer sender)
         {
             var block = message.Payload;
-
+            
             if (block.Hash == null)
             {
-                this._blockOperationsManager.Sign(block);
+                this._blockSigner.Sign(block);
+            }
+
+            if (_blockVerifier.Verify(block))
```

Comment:
I guess we can save a bit here - no need in verification if we just signed a block.

Reply:
I meant if the hash was null on line 58 and we signed on line 59. It makes a little sense to verify on line 62.